entity/c/c_common.h

```diff
@@ -0,0 +1,175 @@
+#ifndef C_COMMON_H
+#define C_COMMON_H
+
+#include <stdio.h>
+#include <stdlib.h>
+#include <stdbool.h>
+#include <arpa/inet.h>
+#include <sys/socket.h>
+#include <sys/time.h>
+#include <sys/types.h>
+#include <netinet/in.h>
+#include <unistd.h>
+#include <arpa/inet.h>
+#include <unistd.h>
+#include <string.h>
+#include <stdint.h>
+#include <time.h>
+#include <math.h>
+#include <openssl/rand.h>
+#include <pthread.h>
+
+// Message Type //
+#define AUTH_HELLO 0
+#define ENTITY_HELLO 1
+#define AUTH_SESSION_KEY_REQ 10
+#define AUTH_SESSION_KEY_RESP 11
+#define SESSION_KEY_REQ_IN_PUB_ENC 20
+#define SESSION_KEY_RESP_WITH_DIST_KEY 21 // Includes distribution message (session keys)
+#define SESSION_KEY_REQ 22                // Distribution message
+#define SESSION_KEY_RESP 23               // Distribution message
+#define SKEY_HANDSHAKE_1 30
+#define SKEY_HANDSHAKE_2 31
+#define SKEY_HANDSHAKE_3 32
+#define SECURE_COMM_MSG 33
+#define FIN_SECURE_COMM 34
+#define SECURE_PUB 40
+#define MIGRATION_REQ_WITH_SIGN 50
+#define MIGRATION_RESP_WITH_SIGN 51
+#define MIGRATION_REQ_WITH_MAC 52
+#define MIGRATION_RESP_WITH_MAC 53
+#define AUTH_ALERT 100
+
+// Size //
+#define MESSAGE_TYPE_SIZE 1
+#define MAX_PAYLOAD_BUF_SIZE 5
+#define HS_NONCE_SIZE 8
+#define HS_INDICATOR_SIZE 1 + HS_NONCE_SIZE * 2
+
+#define SEQ_NUM_SIZE 8
+
+// Auth Hello //
+#define AUTH_ID_LEN 4
+#define NUMKEY_SIZE 4
+#define NONCE_SIZE 8
+
+// Session key Resp //
+#define MAC_SIZE 32
+#define KEY_ID_SIZE 8
+
+// Handshake struct including nonce, reply_nonce(received),
+// and Diffie Helman parameter
+
+typedef struct
+{
+    unsigned char nonce[HS_NONCE_SIZE];
+    unsigned char reply_nonce[HS_NONCE_SIZE];
+    unsigned char dhParam[HS_NONCE_SIZE]; // TODO: check_size.
```

Comment:
What is this TODO for?

Reply:
DO NOT resolve a conversation without addressing the comments or responding to the comment properly.